hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AReadOpContext.java

```diff
@@ -77,11 +80,13 @@ public S3AReadOpContext(
       Invoker invoker,
       @Nullable FileSystem.Statistics stats,
       S3AStatisticsContext instrumentation,
-      FileStatus dstFileStatus) {
+      FileStatus dstFileStatus,
+      final IOStatisticsContext ioStatisticsContext) {
 
     super(invoker, stats, instrumentation,
         dstFileStatus);
     this.path = requireNonNull(path);
+    this.ioStatisticsContext = ioStatisticsContext;
```

Comment:
can this be null? prefer not

Reply:
Adding EmptyIOStatisticsContext for the disabled scenario with EmptyIOStatisticsStore as the aggregator which is a no-op.